dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -74,261 +67,222 @@ void PropertyEntranceComponent::OnEnterProperty(Entity* entity, uint32_t index,
-		auto friendCheck = Database::CreatePreppedStmt("SELECT best_friend FROM friends WHERE (player_id = ? AND friend_id = ?) OR (player_id = ? AND friend_id = ?)");
+	auto charQuery = Database::CreatePreppedStmt("SELECT name FROM charinfo WHERE prop_clone_id = ?;");
+	charQuery->setUInt64(1, entry.CloneID);
 
-		friendCheck->setUInt(1, character->GetID());
-		friendCheck->setUInt(2, ownerObjId);
-		friendCheck->setUInt(3, ownerObjId);
-		friendCheck->setUInt(4, character->GetID());
+	auto charResult = charQuery->executeQuery();
 
-		auto friendResult = friendCheck->executeQuery();
+	if (charResult->next()) {
+		entry.PrimaryData.OwnerName = std::string(charResult->getString(1).c_str());
+	}
 
-		// If we got a result than the two players are friends.
-		if (friendResult->next()) {
-			isFriend = true;
-			if (friendResult->getInt(1) == 3) {
-				isBestFriend = true;
-			}
-		}
+	delete charResult;
+	charResult = nullptr;
 
-		delete friendCheck;
-		friendCheck = nullptr;
+	delete charQuery;
+	charQuery = nullptr;
 
-		delete friendResult;
-		friendResult = nullptr;
+	entry.PrimaryData.OwnerID = result->getInt(2);
+	entry.PrimaryData.IsModeratorApproved = result->getBoolean(10);
 
-		bool isModeratorApproved = propertyEntry->getBoolean(10);
+	entry.PrimaryData.Name = std::string(result->getString(5).c_str());
+	entry.PrimaryData.Description = std::string(result->getString(6).c_str());
+	entry.PrimaryData.Reputation = result->getUInt(14);
 
-		if (!isModeratorApproved && entity->GetGMLevel() >= eGameMasterLevel::LEAD_MODERATOR) {
-			propertyName = "[AWAITING APPROVAL]";
-			propertyDescription = "[AWAITING APPROVAL]";
-			isModeratorApproved = true;
-		}
+	entry.MetaData.AccessType = result->getInt(9);
+	entry.MetaData.DateLastPublished = result->getInt64(11);
+	entry.MetaData.PerformanceCost = result->getFloat(16);
 
-		bool isAlt = false;
-		// Query to determine whether this property is an alt character of the entity.
-		auto isAltQuery = Database::CreatePreppedStmt("SELECT id FROM charinfo where account_id in (SELECT account_id from charinfo WHERE id = ?) AND id = ?;");
+	delete result;
+	result = nullptr;
 
-		isAltQuery->setInt(1, character->GetID());
-		isAltQuery->setInt(2, owner);
+	delete propQuery;
+	propQuery = nullptr;
 
-		auto isAltQueryResults = isAltQuery->executeQuery();
+	this->m_PropertyDataCache[entry.CloneID] = entry;
 
-		if (isAltQueryResults->next()) {
-			isAlt = true;
-		}
+	return entry;
+}
 
-		delete isAltQueryResults;
-		isAltQueryResults = nullptr;
+PropertyPersonalData PropertyEntranceComponent::GetPropertyPersonalData(PropertyData& propertyData, Entity* queryingUser, bool updatePropertyDataStructure) {
+	PropertyPersonalData personalData{};
 
-		delete isAltQuery;
-		isAltQuery = nullptr;
+	personalData.IsFriend = m_UserFriendMap[(uint32_t)queryingUser->GetObjectID()].find(propertyData.PrimaryData.OwnerID) != m_UserFriendMap[(uint32_t)queryingUser->GetObjectID()].end();
```

Comment:
If the above logic follows, this should attempt to fill the user friend map if the user is not found in it correct?  would be a good safeguard for future users unless this is a private method upon which this can be dismissed.  Same goes for the other new methods; can these be made private/const if possible?